csrc/ir/utils.cpp

```diff
@@ -1366,46 +1366,52 @@ void verifyMmaOpForEvaluation(
       "MmaOp inputs should be the same dtype as the output dtype of the final castOp.");
 }
 
-bool matchMatmulCast(const UnaryOp* cast_op, Val*& mma_lhs, Val*& mma_rhs) {
-  auto* mma = dynamic_cast<MmaOp*>(cast_op->input(0)->definition());
-  if (mma == nullptr) {
-    return false;
-  }
-  MmaOpUtils::verifyMmaOpForEvaluation(
-      mma, cast_op->out()->getDataType().value());
-  mma_lhs = mma->inA();
-  mma_rhs = mma->inB();
-  return true;
-}
-
 // Possible combinations:
 // 1. A x B + C
 // 2. alpha * A x B + C
 // 3. A x B + beta * C
 // 4. alpha * A x B  + beta * C
-bool matchMatmulBiasCast(
+// 5. A x B
+// 6. alpha * A x B
+// Note: We assume the first operand to be the MmaOp output
+bool matchMatmulPatterns(
     const UnaryOp* cast_op,
     Val*& mma_lhs,
     Val*& mma_rhs,
     Val*& bias,
     Val*& alpha,
     Val*& beta) {
+  // Check if there may be a bias present.
```

Comment:
You might want to either initialize all the outputs to `nullptr` here, or (possibly better) return a struct containing these values. That way you can be sure that checking for `nullptr` really means that element was not matched.

Reply:
Another advantage of using a struct is that you don't have to update every call site if you extend the pattern in the future.

Reply:
Great point, added `MatmulInputs` to hold all the variables which are initialized to `nullptr`.